pkgs/by-name/up/uppaal/package.nix

```
@@ -0,0 +1,88 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  makeWrapper,
+  jdk17,
+  makeDesktopItem,
+  copyDesktopItems,
+  wrapGAppsHook,
+  shared-mime-info,
+  gdk-pixbuf,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "uppaal";
+  version = "5.0";
+  subversion = "0";
+  platform = "linux64";
+
+  src = fetchzip {
+    url = "https://download.${pname}.org/${pname}-${version}/${pname}-${version}.${subversion}/${pname}-${version}.${subversion}-${platform}.zip";
+    hash = "sha256-o71mP2/sDNRpmA1Qx59cvx6t4pk5pP0lrn1CogN3PuM=";
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "uppaal";
+      exec = "uppaal %U";
+      icon = "uppaal";
+      comment = "real-time modelling and verification tool";
+      desktopName = "Uppaal";
+      genericName = "Uppaal";
+      categories = [ "Development" ];
+    })
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    jdk17
+    gdk-pixbuf
+    shared-mime-info
+  ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib/uppaal
+    mkdir -p $out/share/icons/hicolor/16x16/apps
+    mkdir -p $out/share/icons/hicolor/32x32/apps
+    mkdir -p $out/share/icons/hicolor/48x48/apps
+    mkdir -p $out/share/icons/hicolor/64x64/apps
+    mkdir -p $out/share/icons/hicolor/96x96/apps
+    mkdir -p $out/share/icons/hicolor/128x128/apps
+
+    cp res/icon-16x16.png $out/share/icons/hicolor/16x16/apps/uppaal.png
+    cp res/icon-32x32.png $out/share/icons/hicolor/32x32/apps/uppaal.png
+    cp res/icon-48x48.png $out/share/icons/hicolor/48x48/apps/uppaal.png
+    cp res/icon-64x64.png $out/share/icons/hicolor/64x64/apps/uppaal.png
+    cp res/icon-96x96.png $out/share/icons/hicolor/96x96/apps/uppaal.png
+    cp res/icon-128x128.png $out/share/icons/hicolor/128x128/apps/uppaal.png
```

Maybe this could be condensed a bit? Not a blocker though

```suggestion
    for size in 16 32 48 64 96 128; do
      install -Dm444 res/icon-"$size"x"$size".png "$out"/share/icons/hicolor/"$size"x"$size"/apps/uppaal.png
    done
```